crates/puffin/src/lib.rs

```diff
@@ -23,3 +23,8 @@
 
 mod compression;
 pub use compression::CompressionCodec;
+
+mod metadata;
```

Comment:
Note that I have deliberately chosen to not expose anything at the moment. 
I want to create a separate, intentional PR to  add `pub` modifiers after the reader and writer implementations have been added.
This way, we can develop inside this crate without having to worry about breaking changes until we're ready.

Reply:
That's great!